travis/setup_dependencies_common.sh

```diff
@@ -313,7 +321,55 @@ if [[ ! -z $CONDA_DEPENDENCIES ]]; then
     fi
 fi
 
+if [[ ! -z $CONDA_DEPENDENCIES ]]; then
+    # Do a dry run of the conda install here to make sure that pins are
+    # ACTUALLY being respected. This will become unnecessary when
+    # https://github.com/conda/conda/issues/9052
+    # is fixed
+    dry_run_bad=$(conda install --dry-run $CONDA_DEPENDENCIES 2>&1 | grep -c "conflicts with explicit specs")
+    echo "dry_run_bad is $dry_run_bad"
+    if [[ $dry_run_bad -gt 0 ]]; then
```

Comment:
rather than having `dry_run_bad` defined, the usual logic all over this file is to compare it with test -z:

``if [[ ! -z $(conda install --dry-run $CONDA_DEPENDENCIES 2>&1 | grep  "conflicts with explicit specs") ]]; then 

not sure whether it's clearer or not though, and I have to admit this file is my worst pile of hacks ever that just keep growing organically 🙈